in-progress/7588-spartan-clusters.md

```diff
@@ -0,0 +1,301 @@
+    nodePort: 8080
+  logLevel: 'debug'
+  debug: 'discv5:*,aztec:*'
+  fakeProofs: false
+  resources: {}
+
+pxe:
+  replicas: 3
+  service:
+    type: ClusterIP
+    port: 8080
+    targetPort: 8080
+  fakeProofs: false
+  resources: {}
+
+chaos:
+  scenarios:
+    - name: 'boot-node-failure'
+      type: 'pod'
+      selector:
+        matchLabels:
+          app: boot-node
+      action: 'kill'
+      actionOptions:
+        force: true
+```
+
+The `yarn-projects/+end-to-end-base` earthly image will be updated to include `kubectl`, `helm`, and `kind`.
+
+We will add new tests to `yarn-projects/end-to-end`, and expose them to CI as earthly targets in the form `yarn-projects/end-to-end/+network-{test_name}`.
+
+The `ci` github action workflow will have a new job analogous to `e2e` and `bench-e2e` called `network-e2e`. This job will be a matrix on the `+network` targets, and for each it will ensure the test machine has a k8s cluster running with the relevant docker images loaded (via kind), and run a particular earthly target.
+
+Each target will:
+1. deploy a network to the k8s cluster (via helm install)
+2. (optional) apply some chaos configurations
+3. run helm tests
+4. tear down the network
+
+
+## Implementation
+
+### Helm Chart
+
+**anvil**
+
+There will be a deployment for anvil. It will have a single replica, and be exposed via ClusterIP.
+
+**boot node**
+
+There will be a statefulset with a single replica for the boot node. As part of its init container it will deploy the enshrined L1 contracts. Other nodes in the network will be able to resolve the boot node's address via its stable DNS name, e.g. `boot-node-0.aztec-network.svc.cluster.local`.
+
+**full node**
+
+There will be a statefulset for the full nodes. The number of replicas will be configurable. Each full node will have a service exposing its p2p port and node port.
+
+As part of their init container, they will get config from the boot node, including its ENR (which will require exposing this on the `get-node-info` endpoint). 
+
+Tests will produce a validator set from the full nodes.
+
+It will be possible to address full nodes individually via their stable DNS name, e.g. `full-node-0.aztec-network.svc.cluster.local`, as well as collectively via a service, e.g. `full-node.aztec-network.svc.cluster.local`.
+
+**pxe**
+
+PXEs will be deployed as a statefulset. The number of replicas will be configurable. Each PXE will have a service exposing its port.
+
+PXEs will use the collective full node service as their node url by default.
+
+PXEs will only be able to be addressed individually.
+
+**opentel**
+
+There will be a deployment for opentel. It will have a single replica. Nodes in the network will push their logs to opentel.
+
+**prometheus**
+
+There will be a deployment for prometheus. It will have a single replica.
+
+**grafana**
+
+There will be a deployment for graphana. It will have a single replica, and be exposed via ClusterIP.
+
+
+### Staging Network
```

Comment:
If we go down this route, I'd use this to deploy all of our networks (ie alphanet, devnet, and provernet), not just sequencer nets.

Reply:
For sure. I'd like to demonstrate it working in isolation with this one network and then migrate the others once we have confidence.